pkgs/development/python-modules/bsddb3/default.nix

```
@@ -36,6 +36,17 @@ buildPythonPackage rec {
     export BERKELEYDB_DIR=${pkgs.db.dev};
   '';
 
+  postPatch = ''
+    substituteInPlace test3.py \
+      --replace-fail "from distutils.util import get_platform" "import platform" \
+      --replace-fail "get_platform()" "f'{platform.system().lower()}-{platform.machine()}'" \
```

I just did some detective work and it looks like the testing code broke around 10000 commits ago, because the build directory structure generated by python changed, and since then `checkPhase` doesn't do anything, it only prints: "Ran 0 tests"

The following changes re-enables testing by using the proper imports:
```suggestion
      --replace-fail "from distutils.util import get_platform" "from sysconfig import get_platform" \
      --replace-fail "sys.version[0:3]" "sys.implementation.cache_tag" \
```

---

Thanks a lot for your investigation of the root problem and this proposed fix.